pkg/operation/botanist/kubeapiserver.go

```diff
@@ -323,6 +323,10 @@ func (b *Botanist) computeKubeAPIServerServiceAccountConfig(ctx context.Context,
 		out.Issuer = *config.ServiceAccountConfig.Issuer
 	}
 
+	if config.ServiceAccountConfig.AcceptedIssuers != nil {
+		out.AcceptedIssuers = append([]string{}, config.ServiceAccountConfig.AcceptedIssuers...)
+	}
```

Comment:
Why not simply assigning `out.AcceptedIssuers = config.ServiceAccountConfig.AcceptedIssuers`?

Reply:
My thought here was that it seems more robust to have different backing arrays for the `out.AcceptedIssuers` and `config.ServiceAccountConfig.AcceptedIssuers` slices.

Reply:
Why?

Reply:
Documentation of `b.Shoot.GetInfo()` states that the method should be used only for reading the data of the shoot resource. This is why it seems more logical to me to use a different underlying array since I do not expect modifying the `out` object to actually change the shoot's spec object.

Reply:
OK, but we don't do this anywhere else, so I would still vote for going the straight-forward way for consistency reasons.

Reply:
I get your point and I did the requested changes. The behavior we are supporting is still rather confusing for me. I am leaving a sample test for reference.

```go
It("Failing test", func() {
	botanist.Shoot.SetInfo(&gardencorev1beta1.Shoot{
		Spec: gardencorev1beta1.ShootSpec{
			Kubernetes: gardencorev1beta1.Kubernetes{
				KubeAPIServer: &gardencorev1beta1.KubeAPIServerConfig{
					ServiceAccountConfig: &gardencorev1beta1.ServiceAccountConfig{
						AcceptedIssuers: []string{"a", "b"},
					},
				},
			},
		},
	})

	serviceAccountConfig, err := botanist.computeKubeAPIServerServiceAccountConfig(ctx, botanist.Shoot.GetInfo().Spec.Kubernetes.KubeAPIServer, "something.here.com")
	Expect(err).NotTo(HaveOccurred())
	serviceAccountConfig.AcceptedIssuers[0] = "c"
	Expect(botanist.Shoot.GetInfo().Spec.Kubernetes.KubeAPIServer.ServiceAccountConfig.AcceptedIssuers[0]).To(Equal("a"))
})
```